docs/adr/0006-url-encode-path-parameters.md

```diff
@@ -0,0 +1,51 @@
+# 0006 - URL-encode path parameters that contain forward slashes
+
+2023-03-16
+
+## Status
+
+Accepted
+
+## Context
+
+Users must be able to search the system for nominals on Police National Computer (PNC) ID. PNC ID is a unique person
+identifier. RESTful API design convention is that path parameters are used to identify a specific resource or resources.
+However, passing PNC IDs into our API as path parameters is problematised by the fact that PNC IDs contain a forward
+slash (e.g., 12/345345B) which is the URO standard path delimiter.
+
+We considered two options for dealing with the forward slash in PNC ID:    
+**1. Encode the PNC ID**
+
+- The PNC ID 12/345345B would be passed to the /persons end-point URL-encoded as 12%2F345345B.
+
+Considerations:
+
+- This would result in some added complexity for the consumer, who would have to encode the PNC ID.
+
+**2. Use query parameters to pass the PNC ID to the API**
+
+- The request format would become /persons?pnc_id={pncId} rather than /persons/{pncId}.
+
+Considerations:
+
+- Query parameters should be used to sort or filter on resources and so this usage is not best practice.
```

Comment:
I'm not sure I agree with this comment, we're using query parameters on our search endpoint to search for people by first and last name. You can definitely use query params to control pagination and sorting for sure, but I don't think its necessarily search on them. Perhaps the take away here is that when searching on a single criteria such as an identifier the preferred method would be to do it via the url path.